build.ps1

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env pwsh
+param(
+    [Parameter()]
+    [switch]
+    $Clean,
+
+    [Parameter()]
+    [switch]
+    $BootstrapBuildEnv
+)
+
+$NeededTools = @{
+    OpenSsl = "openssl for macOS"
+    DotNet451TargetingPack = ".NET 4.5.1 Targeting Pack"
+    PowerShellGet = "PowerShellGet latest"
+    InvokeBuild = "InvokeBuild latest"
+}
+
+if ((-not $PSVersionTable["OS"]) -or $PSVersionTable["OS"].Contains("Windows")) {
+    $OS = "Windows"
+} elseif ($PSVersionTable["OS"].Contains("Darwin")) {
+    $OS = "macOS"
+} else {
+    $OS = "Linux"
+}
+
+
+function needsOpenSsl () {
+    if ($OS -eq "macOS") {
+        try {
+            $opensslVersion = (openssl version)
+        } catch {
+            return $true
+        }
+    }
+    return $false
+}
+
+function needsDotNet451TargetingPack () {
+    # how could we check for this?
+    return $false
```

Comment:
`Get-CimInstance Win32_Product | Where-Object Name -match '\.NET Framework 4\.5\.1 Multi-Targeting Pack'` 

However, this call is slow - ~30 secs on my laptop.  We might want to do this check only during bootstrap.  Or we find a quicker way to test for this requirement.

Reply:
That sounds good. I have it running on only Bootstrap and also write to the console a small message that says the check wasn't made when on windows and building.

Reply:
@tylerl0706 part of why it's slow is it verifies and repairs all installed MSI packages, potentially breaking installed applications.  Here's a [kb](https://support.microsoft.com/en-us/help/974524/event-log-message-indicates-that-the-windows-installer-reconfigured-al) on it.  We should definitely find an alternative, maybe searching `HKLM:\SOFTWARE` instead.

Reply:
I did my best to look around for _something_ that could indicate the targeting pack in `HKLM:\SOFTWARE` but I couldn't find anything :(

Reply:
If you have any other hints/pointers, I'm happy to try.

Reply:
https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed#net_b